envoy/event/dispatcher.h

```diff
@@ -286,6 +289,28 @@ class Dispatcher : public DispatcherBase, public ScopeTracker {
    * Shutdown the dispatcher by clear dispatcher thread deletable.
    */
   virtual void shutdown() PURE;
+
+  /**
+   * Provides filters access to connection handler to save outgoing connections as
+   * incoming connections for reverse tunnels
+   */
+  virtual void setConnectionHandler(Network::ConnectionHandler* connection_handler) PURE;
```

Comment:
I don't see what this function has to do with the dispatcher - did you stash this here as a convenient way to get the handler local to your thread?  I think you want to look into the thread local state (tls) getters and setters.